XAMLator.Server.Shared/MonoEvaluator.cs

```diff
@@ -10,12 +10,29 @@ public class Evaluator : IEvaluator
 	{
 		Mono.CSharp.Evaluator eval;
 		Printer printer;
+		bool canEvaluate = false;
 
 		public Task<bool> EvaluateExpression(string expression, string code, EvalResult result)
 		{
 			EnsureConfigured();
 			try
 			{
+				//on real devices we cannot do eval.Evaluate! https://github.com/mono/mono/issues/6616
```

Comment:
I will copy the comments from the previous commit here since they where lost.

I decided to pass the expression string because the type might not have a default constructor. you might for instance want to do things like "new View (<FullXamlStringHere>);"

I think the logic of creating the new type if the evaluation is not possible should be delegated to the VM. I would change the IEvaluator interface to provide a new method that defines whether the platform supports dynamic code generation or not, for example IEvaluator.IsEvaluationSupported ()
To be able to create the new instance without having to parse the evaluation string I would do the following:
    * Add a new field to EvalRequest named OriginalTypeName that includes the full name of the Type.
     * In VM.cs, if the platform does not support evaluation, create a new instance by type as you are doing here but using the OriginalTypeName.

Reply:
Agreed, that would be better. I need to figure out how the plugin-part for VS Mac works first before I can start debugging that though :)